thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -0,0 +1,608 @@
+    if (anomalyFunctionSpec.getIsActive()) {
+      detectorHttpUtils.disableAnomalyFunction(String.valueOf(id));
+    }
+
+    CollectionSchema schema = CACHE_REGISTRY_INSTANCE.getCollectionSchemaCache().get(dataset);
+    TimeGranularity dataGranularity = schema.getTime().getDataGranularity();
+
+    anomalyFunctionSpec.setIsActive(false);
+    anomalyFunctionSpec.setCollection(dataset);
+    anomalyFunctionSpec.setFunctionName(functionName);
+    anomalyFunctionSpec.setMetric(metric);
+    if (StringUtils.isEmpty(type)) {
+      type = DEFAULT_FUNCTION_TYPE;
+    }
+    anomalyFunctionSpec.setType(type);
+    anomalyFunctionSpec.setWindowSize(Integer.valueOf(windowSize));
+    anomalyFunctionSpec.setWindowUnit(TimeUnit.valueOf(windowUnit));
+    anomalyFunctionSpec.setWindowDelay(Integer.valueOf(windowDelay));
+    if (StringUtils.isEmpty(windowDelayUnit)) {
+      anomalyFunctionSpec.setWindowDelayUnit(TimeUnit.valueOf(windowUnit));
+    } else {
+      anomalyFunctionSpec.setWindowDelayUnit(TimeUnit.valueOf(windowDelayUnit));
+    }
+    anomalyFunctionSpec.setBucketSize(dataGranularity.getSize());
+    anomalyFunctionSpec.setBucketUnit(dataGranularity.getUnit());
+    anomalyFunctionSpec.setExploreDimensions(exploreDimensions);
+    anomalyFunctionSpec.setProperties(properties);
+
+    String cron = DEFAULT_CRON;
+    if (StringUtils.isNotEmpty(repeatEvery)) {
+      cron = ThirdEyeUtils.constructCron(scheduleMinute, scheduleHour, repeatEvery);
+    }
+    anomalyFunctionSpec.setCron(cron);
+
+    Long responseId = anomalyFunctionSpecDAO.createOrUpdate(anomalyFunctionSpec);
+
+    if (isActive) {
+      detectorHttpUtils.enableAnomalyFunction(String.valueOf(responseId));
+    }
+
+    return Response.ok(responseId).build();
+  }
+
+  // Delete anomaly function
+  @DELETE
+  @UnitOfWork
+  @Path("/anomaly-function/delete")
+  public Response deleteAnomalyFunctions(@QueryParam("id") Long id, @QueryParam("functionName") String functionName)
+      throws IOException {
+
+    // call endpoint to stop if active
+    AnomalyFunctionSpec anomalyFunctionSpec = anomalyFunctionSpecDAO.findById(id);
+    if (anomalyFunctionSpec.getIsActive()) {
+      detectorHttpUtils.disableAnomalyFunction(String.valueOf(id));
+    }
+
+    // delete from db
+    anomalyFunctionSpecDAO.delete(anomalyFunctionSpec);
+
+    return Response.noContent().build();
+  }
+
+  // Run anomaly function ad hoc
+  @POST
+  @UnitOfWork
+  @Path("/anomaly-function/adhoc")
+  public Response runAdhocAnomalyFunctions(@QueryParam("id") Long id, @QueryParam("functionName") String functionName,
+      @QueryParam("windowStartIso") String windowStartIso, @QueryParam("windowEndIso") String windowEndIso)
+          throws Exception {
+
+    AnomalyFunctionSpec anomalyFunctionSpec = anomalyFunctionSpecDAO.findById(id);
+    if (StringUtils.isEmpty(windowStartIso) || StringUtils.isEmpty(windowEndIso)) {
+      int windowSize = anomalyFunctionSpec.getWindowSize();
```

Comment:
no need to do now()-based logic here, just construct url via httpl util and pass in the query params for start/end if they are present.